source/slang/check.cpp

```diff
@@ -1291,17 +1293,17 @@ namespace Slang
         {
             if(left == right) return true;
 
-            if(auto leftConst = left.As<ConstantIntVal>())
+            if(auto leftConst = left.dynamicCast<ConstantIntVal>())
```

Comment:
As I'm wading through these diffs, I am having two thoughts about how we might have been able to stage this to make the reviews simpler:

1. We could have started with the name `As` instead of `as` just to keep things simple.

2. For cases like this where the difference doesn't matter, we could have favored `As` over `dynamicCast`

That would have kept the diffs as small as possible (really just renaming `dynamic_cast` to `dynamicCast` and any existing uses of true dynamic cast on `Type` over to `dynamicCast`), and then do a separate PR for a bulk rename of `As` to `as` (that wouldn't need a line-by-line review).

Not a criticism of this PR, but just a thought about how we can try to stage whole-codebase changes like this to be simpler to review.